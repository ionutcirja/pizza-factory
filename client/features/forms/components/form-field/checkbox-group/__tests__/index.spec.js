import React from 'react';
import { shallow } from 'enzyme';
import { CheckboxGroupField } from '..';

describe('CheckboxGroupField component', () => {
  let propsToRender;
  
  beforeEach(() => {
    propsToRender = {
      field: {
        name: 'option',
        value: [],
      },
      list: [
        {
          value: 1,
        },
        {
          value: 2,
        },
        {
          value: 3,
        },
        {
          value: 4,
        },
      ],
      maxAllowed: 3,
      form: {
        touched: {},
        errors: {},
        setFieldValue: jest.fn(),
      },
      theme: {
        colours: {
          grey: 'grey',
          black: 'black',
          red: 'red',
          blue: 'blue',
        },
      },
    };
  });
  
  describe('render', () => {
    it('should render a list of checkboxes', () => {
      const wrapper = shallow(<CheckboxGroupField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
    
    it('should render check checkboxes if their values are already in the field values list', () => {
      propsToRender.field.value = ['1', '2'];
      const wrapper = shallow(<CheckboxGroupField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render some disabled checkboxes if their values are not in the field values list'
      + ' and the maximum allowed checked checkboxes number is already met', () => {
      propsToRender.field.value = ['1', '2', '3'];
      const wrapper = shallow(<CheckboxGroupField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
    
    it('should render an error if field is touched and'
      + ' there is an error defined for the current field', () => {
      propsToRender.form = {
        touched: {
          option: true,
        },
        errors: {
          option: 'some error',
        },
      };
      const wrapper = shallow(<CheckboxGroupField {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  
  describe('events handlers', () => {
    it('should add a checkbox value into the values list on change if it is not already contained', () => {
      propsToRender.field.value = ['1'];
      const wrapper = shallow(<CheckboxGroupField {...propsToRender} />);
      wrapper.find('Checkbox').at(2).props().onChange({ target: { value: '2' } });
      expect(propsToRender.form.setFieldValue).toHaveBeenCalledWith('option', ['1', '2']);
    });
  
    it('should remove a checkbox value from the values list on change if it is already contained', () => {
      propsToRender.field.value = ['1', '2'];
      const wrapper = shallow(<CheckboxGroupField {...propsToRender} />);
      wrapper.find('Checkbox').at(1).props().onChange({ target: { value: '1' } });
      expect(propsToRender.form.setFieldValue).toHaveBeenCalledWith('option', ['2']);
    });
  });
});
