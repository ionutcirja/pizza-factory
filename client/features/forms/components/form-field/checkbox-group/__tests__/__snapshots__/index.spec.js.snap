// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroupField component render should render a list of checkboxes 1`] = `
<FieldSet>
  <Label
    as="legend"
    colour="grey"
  />
  <div
    key="1"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      id="option.0"
      name="option.0"
      onChange={[Function]}
      type="checkbox"
      value={1}
    />
    <Label
      colour="grey"
      htmlFor="option.0"
    >
      1
    </Label>
  </div>
  <div
    key="2"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      id="option.1"
      name="option.1"
      onChange={[Function]}
      type="checkbox"
      value={2}
    />
    <Label
      colour="grey"
      htmlFor="option.1"
    >
      2
    </Label>
  </div>
  <div
    key="3"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      id="option.2"
      name="option.2"
      onChange={[Function]}
      type="checkbox"
      value={3}
    />
    <Label
      colour="grey"
      htmlFor="option.2"
    >
      3
    </Label>
  </div>
  <div
    key="4"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      id="option.3"
      name="option.3"
      onChange={[Function]}
      type="checkbox"
      value={4}
    />
    <Label
      colour="grey"
      htmlFor="option.3"
    >
      4
    </Label>
  </div>
</FieldSet>
`;

exports[`CheckboxGroupField component render should render an error if field is touched and there is an error defined for the current field 1`] = `
<FieldSet>
  <Label
    as="legend"
    colour="grey"
  />
  <div
    key="1"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      error="some error"
      id="option.0"
      name="option.0"
      onChange={[Function]}
      type="checkbox"
      value={1}
    />
    <Label
      colour="grey"
      htmlFor="option.0"
    >
      1
    </Label>
  </div>
  <div
    key="2"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      error="some error"
      id="option.1"
      name="option.1"
      onChange={[Function]}
      type="checkbox"
      value={2}
    />
    <Label
      colour="grey"
      htmlFor="option.1"
    >
      2
    </Label>
  </div>
  <div
    key="3"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      error="some error"
      id="option.2"
      name="option.2"
      onChange={[Function]}
      type="checkbox"
      value={3}
    />
    <Label
      colour="grey"
      htmlFor="option.2"
    >
      3
    </Label>
  </div>
  <div
    key="4"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      error="some error"
      id="option.3"
      name="option.3"
      onChange={[Function]}
      type="checkbox"
      value={4}
    />
    <Label
      colour="grey"
      htmlFor="option.3"
    >
      4
    </Label>
  </div>
  <Error
    colour="red"
  >
    some error
  </Error>
</FieldSet>
`;

exports[`CheckboxGroupField component render should render check checkboxes if their values are already in the field values list 1`] = `
<FieldSet>
  <Label
    as="legend"
    colour="grey"
  />
  <div
    key="1"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={true}
      disabled={false}
      id="option.0"
      name="option.0"
      onChange={[Function]}
      type="checkbox"
      value={1}
    />
    <Label
      colour="grey"
      htmlFor="option.0"
    >
      1
    </Label>
  </div>
  <div
    key="2"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={true}
      disabled={false}
      id="option.1"
      name="option.1"
      onChange={[Function]}
      type="checkbox"
      value={2}
    />
    <Label
      colour="grey"
      htmlFor="option.1"
    >
      2
    </Label>
  </div>
  <div
    key="3"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      id="option.2"
      name="option.2"
      onChange={[Function]}
      type="checkbox"
      value={3}
    />
    <Label
      colour="grey"
      htmlFor="option.2"
    >
      3
    </Label>
  </div>
  <div
    key="4"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={false}
      id="option.3"
      name="option.3"
      onChange={[Function]}
      type="checkbox"
      value={4}
    />
    <Label
      colour="grey"
      htmlFor="option.3"
    >
      4
    </Label>
  </div>
</FieldSet>
`;

exports[`CheckboxGroupField component render should render some disabled checkboxes if their values are not in the field values list and the maximum allowed checked checkboxes number is already met 1`] = `
<FieldSet>
  <Label
    as="legend"
    colour="grey"
  />
  <div
    key="1"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={true}
      disabled={false}
      id="option.0"
      name="option.0"
      onChange={[Function]}
      type="checkbox"
      value={1}
    />
    <Label
      colour="grey"
      htmlFor="option.0"
    >
      1
    </Label>
  </div>
  <div
    key="2"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={true}
      disabled={false}
      id="option.1"
      name="option.1"
      onChange={[Function]}
      type="checkbox"
      value={2}
    />
    <Label
      colour="grey"
      htmlFor="option.1"
    >
      2
    </Label>
  </div>
  <div
    key="3"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={true}
      disabled={false}
      id="option.2"
      name="option.2"
      onChange={[Function]}
      type="checkbox"
      value={3}
    />
    <Label
      colour="grey"
      htmlFor="option.2"
    >
      3
    </Label>
  </div>
  <div
    key="4"
  >
    <Checkbox
      borderColour="grey"
      borderErrorColour="red"
      checkColour="blue"
      checked={false}
      disabled={true}
      id="option.3"
      name="option.3"
      onChange={[Function]}
      type="checkbox"
      value={4}
    />
    <Label
      colour="grey"
      htmlFor="option.3"
    >
      4
    </Label>
  </div>
</FieldSet>
`;
